osu.Game/Screens/Multi/Components/RoomPollingComponent.cs

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Collections.Generic;
+using osu.Framework.Allocation;
+using osu.Framework.Bindables;
+using osu.Game.Online;
+using osu.Game.Online.API;
+using osu.Game.Online.Multiplayer;
+
+namespace osu.Game.Screens.Multi.Components
+{
+    public abstract class RoomPollingComponent : PollingComponent
+    {
+        public Action<List<Room>> RoomsReceived;
+
+        public readonly Bindable<bool> InitialRoomsReceived = new Bindable<bool>();
```

Comment:
What's the point of this one? I've checked follow up PRs and can only find actual usage in tests (and it feels a bit flaky because it's getting reset by a different class). Thinking maybe an event action may work better if that's the only usage?

Reply:
`RoomManager`'s correspondent bindable binds to these ones I'm pretty sure, and higher up `InitialRoomsReceived` controls the lounge loading layer showing up. I do agree that I'm not 100% on this structure either, given that `RoomPollingComponent` does

```csharp
InitialRoomsReceived.Value = true;
RoomsReceived?.Invoke(rooms);
```

Feels like two members doing kind of the same thing.

Reply:
Aha, right. Found it in `LoungeSubScreen`.

Yeah still not 100% sold on the bindable being exposed (would prefer an `IBindable` + `ResetRooms()` method or similar?). It looks like the rationale for doing it this way was to pass the bindable down one or two levels, so not sure how much overhead that will add without trying to implement it.

Reply:
I did it this way to avoid making unnecessary changes into what is already a large changeset. But sure.